service/raft/group0_state_machine.cc

```diff
@@ -242,7 +243,9 @@ future<> group0_state_machine::merge_and_apply(group0_state_machine_merger& merg
     [&] (schema_change& chng) -> future<> {
         auto modules_to_reload = get_modules_to_reload(chng.mutations);
         co_await _mm.merge_schema_from(locator::host_id{cmd.creator_id.uuid()}, std::move(chng.mutations));
+        if (_in_memory_state_machine_enabled) {
```

Comment:
Can you refactor the visitor to only apply changes and move reloading outside to there will be only one `if()`?